resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/resources/session/TestBug297635.java

```diff
@@ -58,86 +104,73 @@ public void test1() {
 		} catch (IOException e) {
 			fail("1.2", e);
 		}
+	}
 
-		// register a save participant for the bundle
+	private void addSaveParticipant(String failMessage) {
 		try {
 			getWorkspace().addSaveParticipant(BUNDLE01_ID, TestBug297635.this);
 		} catch (CoreException e) {
-			fail("2.0", e);
+			fail(failMessage, e);
 		}
+	}
 
-		// create a project with a file
-		IProject project = getWorkspace().getRoot().getProject("Project1");
-		IFile file = project.getFile("file1.txt");
-		ensureExistsInWorkspace(project, true);
-		ensureExistsInWorkspace(file, getRandomContents());
-
-		// perform a full save
+	private void saveUsingWorkspace() {
 		try {
 			getWorkspace().save(true, getMonitor());
 		} catch (CoreException e) {
 			fail("3.0", e);
 		}
 	}
 
-	@SuppressWarnings("unchecked")
-	public void test2() {
+	private void reinstallBundle() {
 		// install the bundle again
-		// we need to restart the org.eclipse.core.resources bundle to read the tree file again
-		// we rely on the fact that the core.resources bundle doesn't save the tree when it is stopped
+		// we need to restart the org.eclipse.core.resources bundle to read the tree
+		// file again
+		// we rely on the fact that the core.resources bundle doesn't save the tree when
+		// it is stopped
 		try {
 			Bundle coreResourcesBundle = Platform.getBundle(ResourcesPlugin.PI_RESOURCES);
 			coreResourcesBundle.stop(Bundle.STOP_TRANSIENT);
-			Bundle b = BundleTestingHelper.installBundle("1", getContext(), ContentTypeTest.TEST_FILES_ROOT + "content/bundle01");
-			BundleTestingHelper.resolveBundles(getContext(), new Bundle[] {b});
+			Bundle b = BundleTestingHelper.installBundle("1", getContext(),
+					ContentTypeTest.TEST_FILES_ROOT + "content/bundle01");
+			BundleTestingHelper.resolveBundles(getContext(), new Bundle[] { b });
 			coreResourcesBundle.start(Bundle.START_TRANSIENT);
 		} catch (MalformedURLException e2) {
-			fail("1.0", e2);
+			fail("4.0", e2);
 		} catch (BundleException e2) {
-			fail("1.1", e2);
+			fail("4.1", e2);
 		} catch (IOException e2) {
-			fail("1.2", e2);
+			fail("4.2", e2);
 		}
+	}
 
-		// create yet another file in the existing project
-		IProject project = getWorkspace().getRoot().getProject("Project1");
-		IFile file = project.getFile("file2.txt");
-		ensureExistsInWorkspace(file, getRandomContents());
-
-		// get access to SaveManager#savedStates to verify that tress are being kept there
+	@SuppressWarnings("unchecked")
+	private Map<String, SavedState> getSavedStatesFromSaveManager() {
+		// get access to SaveManager#savedStates to verify that tress are being kept
+		// there
 		Map<String, SavedState> savedStates = null;
 		try {
 			Field field = SaveManager.class.getDeclaredField("savedStates");
 			field.setAccessible(true);
 			savedStates = (Map<String, SavedState>) field.get(((Workspace) getWorkspace()).getSaveManager());
 		} catch (IllegalArgumentException e) {
-			fail("2.0", e);
+			fail("5.0", e);
```

Comment:
Instead of fail, why not simply throw the exception (what would fail he test as well...)